scripts/dist-fix.js

```diff
@@ -43,7 +42,7 @@ const path = require('node:path')
         const esmFile = await readFile(distEsmFilePath)
         const esmFileContents = esmFile.toString()
 
-        const dtsFilePath = `./${esmFilePath.replace('.js', '.d.ts')}`
+        const dtsFilePath = `./${esmFilePath.replace('.js', '.d.mts')}`
```

Comment:
I somewhat alluded to this in the commit message: In the current config, `tsup` generates **two** type definition files:
- `index.d.ts`: the type def file for CJS output
- `index.d.mts`: the type def file for ESM output

Before my changes, `index.d.mts` was left unused in the root dist folder, and`index.d.ts` was (1) copied into `dist/esm` and then (2) forces as type reference in the ESM index file. (I suspect this setup could have been due to a bug in a previous `tsup` version, which perhaps did not generate this dedicated `index.d.mts` file?)

Before the split, copying & reusing the CJS type def file was be fine, because the two type def files were identical. However, they are not _guaranteed_ to be identical. Using the dedicated `.d.mts` file is more accurate/"correct", and requires less extra file juggling moving forward with multiple dist files, chunk files, etc.

In order for this to not break the injected type references for Deno, I updated this line to point to the dedicated `.d.mts` file. (Great variable name btw; I would have otherwise not known why that injection was put in place.)